judgments/utils/utils.py

```diff
@@ -186,9 +186,17 @@ def has_filters(query_params, exclude=["order", "per_page"]):
     return len(set(k for (k, v) in query_params.items() if v) - set(exclude)) > 0
 
 
-def get_document_by_uri(document_uri: str) -> Document:
+def get_document_by_uri(document_uri: str, max_ttl: int = 900) -> Document:
+    ttl_hash = round(time() / max_ttl)
+    return get_document_by_uri_from_cache(DocumentURIString(document_uri), ttl_hash=ttl_hash)
+
+
+@lru_cache(maxsize=256)
+def get_document_by_uri_from_cache(document_uri: DocumentURIString, ttl_hash: int = 0) -> Document:
```

Comment:
Docstrings would be appreciated, this is quite subtle. Recommend a link to the SO answer.